packages/hawtio/src/plugins/camel/routes/Source.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { CodeEditor, Language } from '@patternfly/react-code-editor'
+import { CamelContext } from '@hawtiosrc/plugins/camel/context'
+import { MBeanNode } from '@hawtiosrc/plugins/shared'
+
+export const findParentContext = (child: MBeanNode, root: MBeanNode): MBeanNode | undefined => {
+  let found: MBeanNode | undefined
+  root.children?.every(mb => {
+    if (mb.id === child.id) {
+      found = root
+      return false
+    } else found = findParentContext(child, mb)
+    return true
+  })
+
+  return found
+}
+export const Source: React.FunctionComponent = () => {
+  const { selectedNode, tree } = useContext(CamelContext)
+  const [xmlSource, setxmlSource] = useState<string>('')
+
+  useEffect(() => {
+    const xml = selectedNode?.getProperty('xml')
+    if (xml) {
+      setxmlSource(xml)
```

Comment:
Really small thing but should there be an error message applied to `setxmlSource` if xml is null. Obviously, the likelihood it is null will be very small.

Reply:
thanks. I added an error pop up if xml will be null